Add exercise preview cards to DOJO exercises page

bryanjenningz opened the pull request:
This pull request adds a new ExercisePreviewCard component which gets used on the DOJO exercises page.
The exercise preview cards are populated with mock data on the frontend, but this will be removed once the backend returns DOJO exercise preview data.

![screenshot](https://user-images.githubusercontent.com/7637655/189548557-2bbeba5b-84af-4096-9360-8e8cdfecf4b4.png)


This change doesn't affect any user-facing pages since there are no links to the /exercises/:lessonSlug page right now.

How to test:
* Go to /exercises/js0
* Make sure everything looks good

This pull request is a part of https://github.com/garageScript/c0d3-app/issues/2253.

SlyBouhafs commented:
The `answered` state badge looks a bit big, can we make it the same size as the `admin` badge?

bryanjenningz commented:
@SlyBouhafs Thanks for suggestion. I updated the badge size.

flacial commented on components/ExercisePreviewCard.tsx:
```diff
@@ -0,0 +1,43 @@
+}
+
+const ExercisePreviewCard = ({
+  topic,
```
This should be named `moduleName`

bryanjenningz replied:
Done

flacial commented on components/ExercisePreviewCard.tsx:
```diff
@@ -0,0 +1,43 @@
+        <div className={`badge ${topMessageStyle}`}>{state}</div>
+      </div>
+      <div className="mb-2">Problem</div>
+      <pre className="bg-light py-2 px-3">{problem}</pre>
```
In the future, we need to look into highlighting the code. The code doesn't get highlighted in many places such as the comments and lesson/challenge description.

bryanjenningz replied:
I agree, we can use something like react-syntax-highlighter: https://react-syntax-highlighter.github.io/react-syntax-highlighter/demo/

flacial commented on scss/exercisePreviewCard.module.scss:
```diff
@@ -0,0 +1,35 @@
+@use './_variables.module';
+
+%exercisePreviewCard__topBorder {
```
What's this `%`?

bryanjenningz replied:
It marks a block of code that can be extended with `@extend`
It apparently uses up less space in the compiled CSS compared to mixins because mixins will inline the CSS properties, but these will make a reused CSS class.
Look at the "Extend/Inheritance" section on this page: https://sass-lang.com/guide

flacial commented:
We decided to create a folder for each component. This folder will contain the component itself, its styles, and test file.

When creating new components, refer to https://github.com/garageScript/c0d3-app/issues/2080

bryanjenningz commented:
@flacial Thanks for letting me know. I just changed the code so that the component file and styles file are in the same folder. I tried putting the storybook file in the same folder too but it didn't show up when I ran storybook, so I moved the storybook file back to the storybook folder with all the other storybook files.
